### PR TITLE
feat: carousel for resources on gallery detail

### LINKED ISSUE #5603: Make carrousel viewer for multi-resource NFTs
https://canary.kodadot.xyz/ksm/gallery/17383388-3289d9f38266576b1c-AMERICAS2_S01EPISODES-EPISODE_21_SE-00000002
on singular: 
https://singular.app/collectibles/kusama/3289d9f38266576b1c-AMERICAS2_S01EPISODES/17383388-3289d9f38266576b1c-AMERICAS2_S01EPISODES-EPISODE_21_SE-00000004

- there should be more items visible in gallery item detail.

### PR DESCRIPTION
**Pull request by preschian**
**Thank you for your contribution** to the [KodaDot - One Stop Shop for Polkadot NFTs](https://kodadot.xyz).

👇 __ Let's make a quick check before the contribution.

## PR Type

- [ ] Bugfix
- [x] Feature
- [ ] Refactoring

## Context

- [x] Closes #5603

#### Before submitting pull request, please make sure:

- [x] My contribution builds **clean without any errors or warnings**
- [x] I've merged recent default branch -- **main** and I've no conflicts
- [x] I've tried to respect high code quality standards
- [x] I've didn't break any original functionality

#### Optional

- [ ] I've tested it at </ksm/collection>
- [ ] I've tested PR on mobile
- [ ] I've written unit tests 🧪
- [ ] I've found edge cases

#### Had issue bounty label?

- [x] Fill up your KSM address: [Payout](https://beta.kodadot.xyz/transfer/?target=DY4SQF2iD456tH89aQtz5wv1EV3BbSW8wKKuMcwbmXaj1pM)

#### Community participation

- [x] [Are you at KodaDot Discord?](https://discord.gg/35hzy2dXXh)

## Screenshot 📸

- [x] My fix has changed **something** on UI; a screenshot is best to understand changes for others.

Screenshot 2023-04-21 at 18 07 46

### <samp>🤖 Generated by Copilot at f754fbe</samp>

This pull request enhances the gallery item component to show the additional resources of the NFT in a carousel. It introduces a new custom type `NftResources` to represent the resources, and updates the `useGalleryItem` and `useNft` composables to handle the new type. It also modifies the `GalleryItem.vue` component to use the carousel and hide the fullscreen button if there are resources.

### <samp>🤖 Generated by Copilot at f754fbe</samp>

> _To show more resources of an NFT_
> _We added a carousel, you see_
> _We used `useNft`_
> _And changed `ItemResources` a bit_
> _To make the types more consistent and pretty_

### CODE REVIEW

File: components/gallery/GalleryItem.vue [hunk 1]
```diff
@@ -12,12 +12,30 @@
...
             @click="isFullscreen = true">
             <NeoIcon icon="expand" />
           </a>
+          <div v-if="nftResources?.length" class="gallery-media-carousel">
+            <o-carousel
+              :overlay="overlayCarousel"
+              @click="overlayCarousel = !overlayCarousel">
+              <o-carousel-item
+                v-for="(resource, index) in nftResources"
+                :key="index">
```
Comment (Jarsen136) [inline components/gallery/GalleryItem.vue:26]:
```suggestion
                :key="resource.id">
```

**Review by Jarsen136 — COMMENTED** [review]
It looks strange that there is only one dot there.
https://deploy-preview-5751--koda-canary.netlify.app/ksm/gallery/13328844-54bbd380dc3baaa27b-EVRSOULS-Alchemist-00000783
image

**Comment by exezbcz** [comment]
@preschian yup, works

**Review by roiLeo — COMMENTED** [review]
Hey I don't know if it's related but it's missing hover fullscreenn button on media

Screenshot 2023-04-21 at 14-18-57 Wine dogo KodaDot - NFT Market Explorer

**Comment by preschian** [comment]
> Hey I don't know if it's related but it's missing hover fullscreenn button on media

seems like there is no hover style even on canary or production?

File: components/gallery/GalleryItem.vue [hunk 2]
```diff
@@ -268,4 +295,37 @@ $break-point-width: 930px;
 .h-audio {
   height: 70%;
 }
+
+@import '@/styles/abstracts/variables';
```
Comment (Jarsen136) [inline components/gallery/GalleryItem.vue:299]:
```suggestion
```

**Review by Jarsen136 — COMMENTED** [review]
otherwise lgtm

File: components/gallery/GalleryItem.vue (hunk 2 above)
Comment (floyd-li) [reply components/gallery/GalleryItem.vue:299]: > Suggested change
	@import '@/styles/abstracts/variables';

not sure why delete this import? it used `theme` in the following 🤔️

**Comment by floyd-li** [comment]
> > Hey I don't know if it's related but it's missing hover fullscreenn button on media
> 
> seems like there is no hover style even on canary or production?

yes it do have the fullscreen button, i added it lol

image

and this `fullscreen preview` has different behavior with current one. but i think it's okay to merge this pr first, it can done in a separate pr.

https://user-images.githubusercontent.com/16473062/233821520-0bd24769-4179-4b2c-9e4e-bb6e45f44d89.mp4

**Review by floyd-li — APPROVED** [review]
lgtm

File: components/gallery/GalleryItem.vue (hunk 2 above)
Comment (Jarsen136) [reply components/gallery/GalleryItem.vue:299]: > > Suggested change
> > @import '@/styles/abstracts/variables';
> 
> not sure why delete this import? it used `theme` in the following 🤔️

If you look at line 233 on this file, you would find that It's a duplicate import

**Comment by vikiival** [comment]
> noted, will check later. currently, the query seems didn't load on my end. stuck on loading

@preschian there was a service outage 

- https://github.com/kodadot/rubick/issues/256

File: components/gallery/GalleryItem.vue [hunk 3]
```diff
@@ -145,6 +168,10 @@ const tabs = {
 }
 const activeTab = ref(tabs.offers)
 const showCongratsMessage = ref(false)
+const overlayCarousel = ref(false)
+const isResources = computed(
```
Comment (roiLeo) [inline components/gallery/GalleryItem.vue:172]: `isResources` => `hasResources`

File: components/gallery/GalleryItem.vue [hunk 4]
```diff
@@ -12,12 +12,32 @@
...
         <div class="is-relative">
           <a
-            v-if="canPreview"
+            v-if="canPreview && !isResources"
             class="fullscreen-button is-justify-content-center is-align-items-center"
             @click="isFullscreen = true">
             <NeoIcon icon="expand" />
           </a>
+          <div v-if="isResources" class="gallery-item-carousel">
+            <o-carousel
```
Comment (roiLeo) [inline components/gallery/GalleryItem.vue:21]: can't we reuse `CarouselMedia`?

**Review by roiLeo — COMMENTED** [review]
otherwise lgtm

File: components/gallery/GalleryItem.vue (hunk 4 above)
Comment (preschian) [reply components/gallery/GalleryItem.vue:21]: we need to adjust `CarouselMedia` first to disable `nuxt-link` tag

**Review by roiLeo — APPROVED** [review]

**Comment by exezbcz** [comment]
image
- full screen is not working how it should - missing button on hover and it triggers the full screen with a click on the img

**Comment by floyd-li** [comment]
> image
> 
> * full screen is not working how it should - missing button on hover and it triggers the full screen with a click on the img

yes. seems `o-carousel` got a default fullscreen behavior which is different to current. you may take a reference of #5321 i added.

**Comment by exezbcz** [comment]
functionality good - i will follow up with the stylign

**Review by Jarsen136 — APPROVED** [review]

**Comment by vikiival** [comment]
pay 70 usd

**Comment by yangwao** [comment]
😍 Perfect, I’ve sent the payout 
💵 $70 @ 25.52 USD/KSM ~ 2.743 $KSM
🧗 [DY4SQF2iD456tH89aQtz5wv1EV3BbSW8wKKuMcwbmXaj1pM](https://kusama.subscan.io/account/DY4SQF2iD456tH89aQtz5wv1EV3BbSW8wKKuMcwbmXaj1pM)
🔗 [0xafecdedeca3088e06a577137aa6b44d365693d2cbf5e90bf4e3b2c1d89f0028e](https://kusama.subscan.io/extrinsic/0xafecdedeca3088e06a577137aa6b44d365693d2cbf5e90bf4e3b2c1d89f0028e)

🪅 Let’s grab another issue and get rewarded! 
🪄 [github.com/kodadot/nft-gallery/issues](https://github.com/kodadot/nft-gallery/issues/)

**Pull request merged**